### PR TITLE
[Implement-6481] [Task] Add sql task plugin jdbc dependency package

### LINKED ISSUE #6481: [Implement] [Task] Add sql task plugin jdbc dependency package
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

Add sql task plugin jdbc dependency package

### What you expected to happen

Add sql task plugin jdbc dependency package

### How to reproduce

Add sql task plugin jdbc dependency package

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhuangchong**
## Purpose of the pull request
this pr close #6481 

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Comment by CalvinKirs** [comment]
I'm not sure this approach is justified. @dailidong @lenboo PTAL

**Comment by zhuangchong** [comment]
At present, you can choose a variety of known data sources (mysql, pg, ck...) in the sql task. I think the data source currently supported by the system needs to provide jdbc jar. If jdbc jar is not provided, the sql task execution fails. The system is in Before transforming the task plugin, it can support a variety of known data sources jdbc jar.

image

File: dolphinscheduler-task-plugin/dolphinscheduler-task-sql/pom.xml [hunk 1]
```diff
@@ -46,6 +46,274 @@
             <scope>provided</scope>
         </dependency>
 
+        <!-- jdbc dependency -->
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
```
Comment (lenboo) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-sql/pom.xml]: mysql license is not valid for apache-2.0

Comment (zhuangchong) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-sql/pom.xml]: done.

**Pull request closed without merging**